### ISSUE TITLE
Callbacks not processing when using an external object cache

### ISSUE THREAD

**Issue by lakrisgubben** [SwedbankPay/swedbank-pay-woocommerce-payments#41]
Hi,

we've hit some issued with callbacks not processing due to "Unable to get payment ID" on the order when using an external object cache (memcached in this case). The payment id exists in the database but it is not returned when this plugin is using the `get_meta` method on the order object. I managed to track this down to the use of the `save_meta_data` method on `WC_Order`. If one changes all occurences of `save_meta_data` to `save` here and in https://github.com/SwedbankPay/swedbank-pay-woocommerce-core the problem goes away, the reason for this is that `save` seems more aggresive in clearing the caches for the order. And since the `save` method calls the `save_meta_data` [method internally](https://github.com/woocommerce/woocommerce/blob/master/includes/data-stores/abstract-wc-order-data-store-cpt.php#L136) this should hopefully not affect anything else. Two PR's upcoming. Thankful if you could take a look, maybe y'all have a better solution for this. :)

**Comment by lakrisgubben** [comment]
thanks for merging the PR's!

**Comment by lakrisgubben** [comment]
Unfortunately I managed to trigger the same kind of bug even with these changes (though they fixed some issues, but appearently not all :/). I'm thinking this is probably some kind of race condition where the callbacks from SwedbankPay are coming in so quick so the cache has not been busted yet. I found one solution that I've tested quite a few times on staging and it seems to fix this issue. Might not be the most elegant and happy to get feedback on it, but the idea is to force a new db read of WC_Order meta data when getting an order object to bypass the object cache. PR here: https://github.com/SwedbankPay/swedbank-pay-woocommerce-core/pull/15
